### DISCUSSION TITLE
Huge accuracy gap between candle and pytorch on CIFAR-10 simple CNN

### DISCUSSION THREAD

**Discussion by tsubakisakura** [huggingface/candle#2984] [Q&A]
### Issue: Huge Accuracy Gap Between Candle and PyTorch on CIFAR-10 CNN

Hi, I've been comparing Candle and PyTorch implementations of a simple CNN on CIFAR-10. Despite trying to replicate the same architecture and training settings, there's a large accuracy gap between the two:

- PyTorch reaches ~82% train accuracy after 20 epochs.
- Candle stays around ~36% train accuracy, and even starts to degrade after epoch 10.

### Repositories & Code

I've created a minimal public repository for comparison:
https://github.com/tsubakisakura/compare-candle-pytorch

### Implementation

- CNN architecture: 2 conv layers → ReLU → MaxPool → Dropout(0.25) → 2 FC layers
- Loss:
  - PyTorch: `nn.CrossEntropyLoss()`
  - Candle: `candle_nn::loss::cross_entropy(logits, target)`
- Optimizer:
  - PyTorch: `AdamW(lr=1e-3, weight_decay=0.01)`
  - Candle: `AdamW` with same parameters
- Data preprocessing:
  - PyTorch: use torchvision.datasets
  - Candle: use candle_datasets::vision::cifar::load

### Logs

**PyTorch**

~~~
[Epoch 20] Train Loss: 0.4952 Acc: 0.8259 | Test Loss: 0.8723 Acc: 0.7241
~~~

**Candle**

~~~
[Epoch 20] Train Loss: 1.7769 Acc: 0.3603 | Test Loss: 1.7041 Acc: 0.3903
~~~

### Question

I assume Candle is working correctly, but the results differ so much that I'm beginning to doubt whether I may have made a mistake in how I wrote the model or training loop.

Am I missing something in terms of how to correctly express the model or loss function in Candle?
Any suggestions or insight would be greatly appreciated.

Thanks a lot in advance!
